sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -203,8 +203,22 @@ object GpuDeviceManager extends Logging {
       var init = RmmAllocationMode.CUDA_DEFAULT
       val features = ArrayBuffer[String]()
       if (conf.isPooledMemEnabled) {
-        init = init | RmmAllocationMode.POOL
-        features += "POOLED"
+        init = conf.rmmPool match {
+          case c if "default".equalsIgnoreCase(c) =>
+            features += "POOLED"
+            init | RmmAllocationMode.POOL
+          case c if "arena".equalsIgnoreCase(c) =>
+            features += "ARENA"
+            init | RmmAllocationMode.ARENA
+          case c if "none".equalsIgnoreCase(c) =>
+            logWarning(
+              s"RMM pooling is enabled but the pool is set to '$c', pooling is now disabled.")
```

Comment:
This should not warn, this should disable the pool.  Setting the pool mode to `none` should disable pooling even if the deprecated pooling config is true because the latter defaults to true.  We don't want to require users to start using a deprecated config in order to disable the pool.

To be clear, the only behavior I want with respect to the deprecated config is to check if it's set to false (i.e.: NOT the default and therefore intentionally set by the user).  If so, and the pool mode is not "none", the warning should mention the pool enable property is deprecated, but we're overridding the pool mode to be `none` anyway (and may cease to do so in a future release).

Reply:
Done.